game/views.py

```
@@ -3439,3 +3445,59 @@ def download_badge(request, achievement_id):
+        try:
+            move = board.parse_san(move_str)
+            node = node.add_variation(move)
+            board.push(move)
+        except Exception:
+            # Try UCI format as fallback
+            try:
+                move = board.parse_uci(move_str)
+                node = node.add_variation(move)
+                board.push(move)
+            except Exception:
+                break
```

Ruff (0.15.15)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)